nemoguardrails/library/gotitai/actions.py

```diff
@@ -0,0 +1,84 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import logging
+import os
+from typing import Optional
+
+import aiohttp
+
+from nemoguardrails.actions import action
+
+log = logging.getLogger(__name__)
+
+
+@action(name="call gotitai truthchecker api", is_system_action=True)
+async def call_gotitai_truthchecker_api(context: Optional[dict] = None):
+    api_key = os.environ.get("GOTITAI_API_KEY")
+
+    if api_key is None:
+        raise ValueError("GOTITAI_API_KEY environment variable not set.")
+
+    if context is None:
+        raise ValueError(
+            "Context is empty. `user_message`, `bot_response` and `relevant_chunks` keys are required to call the GotIt AI Truthchecker api."
+        )
+
+    user_message = context.get("user_message", "")
+    response = context.get("bot_message", "")
+    knowledge = context.get("relevant_chunks", [])
+
+    if not isinstance(knowledge, list):
+        raise ValueError("`relevant_chunks` must be a list of knowledge.")
```

Comment:
The `relevant_chunks` is always a single string, and this always raises the exception. Try activating the `gotitai rag truthcheck` on the `examples/bots/abc` configuration. In your tests, it works because the `retrieve_relevant_chunks` function is mocked. 

You can split by `\n\n` or use directly `relevant_chunks_sep`.

Reply:
Sounds good! I actually based this on the [factchecking example](https://github.com/NVIDIA/NeMo-Guardrails/blob/d7edd19ddaca0540daefb5dcf01a7b11ecdb1359/nemoguardrails/library/factchecking/align_score/actions.py#L41C5-L41C50). I can adjust it in this PR but I think someone will have to make the change for the `factchecking` example too.

Is the long term plan to do `\n\n` based splitting? I ask because in our internal RAG application, we don't really have this constraint and a single knowledge chunk can have `\n\n` within it.

Reply:
The `\n\n` convention was made early on, and we did not want to break it yet. I suggest going with `relevant_chunks_sep` right now, which means that a chunk can also have `\n\n` in it. When we refactor this (soon), we'll also update the integrations.

Reply:
Thanks! I used `relevant_chunks_sep` in the latest commit.